geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/utils/GeoJsonInference.scala

```diff
@@ -0,0 +1,292 @@
+      val attrMap: java.util.HashMap[String,String] = new java.util.HashMap[String,String](config.asJava)
+      val typeVal = attrMap.get("type").toLowerCase()
+      ad.getType.getBinding match {
+        case t if classOf[java.util.List[_]].isAssignableFrom(t) =>
+          attrMap.put("json-type", "list")
+          attrMap.put("transform", "jsonList('string', $0)")
+          attrMap.put("path", "$.properties." + attrMap.get("name"))
+        case g if classOf[Geometry].isAssignableFrom(g) =>
+          val geomType = attrMap.get("type").toLowerCase()
+          attrMap.put("json-type", "geometry")
+          attrMap.put("transform", s"$geomType($$0)")
+          attrMap.put("path", "$.geometry")
+        case _ =>
+          attrMap.put("json-type", typeVal)
+          attrMap.put("path", "$.properties." + attrMap.get("name"))
+      }
+
+      attrMap.put("name", attrMap.get("name"))
+      attrMap
+    }
+
+    baseConfig
+      .withValue(TypeNamePath, ConfigValueFactory.fromAnyRef(sft.getTypeName))
+      .withValue(FieldsPath, ConfigValueFactory.fromIterable(attributes))
+  }
+
+  def mergeMaps(attrsA: Map[String, AttributeDescriptor],
+                attrsB: Map[String, AttributeDescriptor]): Map[String, AttributeDescriptor] = {
+    val (overlappingKeys, disjointKeys) = attrsB.keys.partition(attrsA.contains)
+
+    // start with fields unique to A
+    var newMap = attrsA.filter( p => !attrsB.contains(p._1)).toSeq
+    // Add fields unique to B
+    disjointKeys.foreach{ name => newMap = newMap :+ (name, attrsB(name)) }
+    // Merge overlaps
+    val merged = overlappingKeys.map { name =>
+      val bindingA = attrsA(name).getType.getBinding
+      val bindingB = attrsB(name).getType.getBinding
+      if (bindingB == classOf[jl.Object] || bindingA == bindingB) {
+        (name, attrsA(name))
+      } else if (bindingA == classOf[jl.Object]) {
+        (name, attrsB(name))
+      } else {
+        val JDouble = classOf[jl.Double]
+        val JInteger = classOf[jl.Integer]
+        val JLong = classOf[jl.Long]
+        (bindingA, bindingB) match {
+          case (JDouble, JInteger) => (name, attrsA(name))
+          case (JInteger, JDouble) => (name, attrsB(name))
+          case (JLong, JInteger) => (name, attrsA(name))
+          case (JInteger, JLong) => (name, attrsB(name))
+          case _ => (name, attrsA(name))
+        }
+      }
+    }
+    newMap = newMap ++ merged
+    newMap.toMap
+  }
+
+  def mergeGeomTypes(geomTypes: Array[String]): String = {
+    val firstNonNull = geomTypes.find(t => t != null)
+    firstNonNull match {
+      case Some(geomType) =>
+        if (geomTypes.forall(_.equals(geomType))) {
+          geomType
+        } else {
+          "Geometry"
+        }
+      case None => throw new UnsupportedOperationException("Could not locate geometry in geojson file.")
+    }
+  }
+
+  // NB: This method is adapted from Apache Spark's JsonInferSchema class
```

Comment:
we probably need to add the apache copyright here

Reply:
@elahrvivaz We'll probably need to add a CQ for redistribution of the source code in this file.

Reply:
does this count as 'redistribution'? we already have a CQ for spark, so I thought adding the copyright would be good enough, as we are 'taking ownership' of this piece of code